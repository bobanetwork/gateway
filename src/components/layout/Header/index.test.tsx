import { render, screen } from '@testing-library/react'
import React from 'react'
import { Header } from '.'
import CustomThemeProvider from 'themes'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import store from 'store'

const renderHeader = () => {
  return render(
    <MemoryRouter>
      <Provider store={store}>
        <CustomThemeProvider>
          <Header />
        </CustomThemeProvider>
      </Provider>
    </MemoryRouter>
  )
}

describe('Layout', () => {
  describe('Header', () => {
    test('should have button with label connect wallet', () => {
      renderHeader()
      expect(screen.getByText('Connect Wallet')).toBeVisible()
    })

    test('should have nav item of length 5 with first element to bridge page.', () => {
      renderHeader()
      const links = screen.getAllByRole('link')
      expect(links.length).toBe(5)
      expect(links[0]).toHaveAttribute('href', '/bridge')
    })
  })
})
