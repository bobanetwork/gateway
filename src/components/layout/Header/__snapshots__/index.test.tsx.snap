// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout => Header should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-fPXMVe ftesQA"
  >
    <a
      class=""
      href="/bridge"
    >
      <div
        class="sc-gFqAkR jmtPti"
      />
    </a>
    <div
      class="sc-hzhJZQ eTHxzX"
    >
      <a
        class="sc-fHjqPf frKNMc ({
              isActive
            }) => {
              /* istanbul ignore next */
              cov_e5m933x91().f[2]++;
              cov_e5m933x91().s[6]++;
              return isActive ?
              /* istanbul ignore next */
              (cov_e5m933x91().b[2][0]++, 'active') :
              /* istanbul ignore next */
              (cov_e5m933x91().b[2][1]++, '');
            }"
        href="/bridge"
      >
        Bridge
      </a>
      <a
        class="sc-fHjqPf frKNMc ({
              isActive
            }) => {
              /* istanbul ignore next */
              cov_e5m933x91().f[2]++;
              cov_e5m933x91().s[6]++;
              return isActive ?
              /* istanbul ignore next */
              (cov_e5m933x91().b[2][0]++, 'active') :
              /* istanbul ignore next */
              (cov_e5m933x91().b[2][1]++, '');
            }"
        href="/history"
      >
        History
      </a>
      <a
        class="sc-fHjqPf frKNMc ({
              isActive
            }) => {
              /* istanbul ignore next */
              cov_e5m933x91().f[2]++;
              cov_e5m933x91().s[6]++;
              return isActive ?
              /* istanbul ignore next */
              (cov_e5m933x91().b[2][0]++, 'active') :
              /* istanbul ignore next */
              (cov_e5m933x91().b[2][1]++, '');
            }"
        href="/earn"
      >
        Earn
      </a>
      <a
        class="sc-fHjqPf frKNMc ({
              isActive
            }) => {
              /* istanbul ignore next */
              cov_e5m933x91().f[2]++;
              cov_e5m933x91().s[6]++;
              return isActive ?
              /* istanbul ignore next */
              (cov_e5m933x91().b[2][0]++, 'active') :
              /* istanbul ignore next */
              (cov_e5m933x91().b[2][1]++, '');
            }"
        href="/stake"
      >
        Stake
      </a>
      <a
        class="sc-fHjqPf frKNMc ({
              isActive
            }) => {
              /* istanbul ignore next */
              cov_e5m933x91().f[2]++;
              cov_e5m933x91().s[6]++;
              return isActive ?
              /* istanbul ignore next */
              (cov_e5m933x91().b[2][0]++, 'active') :
              /* istanbul ignore next */
              (cov_e5m933x91().b[2][1]++, '');
            }"
        href="/DAO"
      >
        Dao
      </a>
    </div>
    <div
      class="sc-ikkxIA fOPkmS"
    >
      <button
        class="sc-cwHptR gLbuCq"
        label="Connect Wallet"
        style="white-space: nowrap;"
        type="button"
      >
         Connect Wallet
      </button>
      <div
        class="sc-dAbbOL cvqTyF"
      >
        <div
          class="sc-feUZmu jPLZfu"
        >
          <div>
            <div
              data-src="hamburger.svg"
              src=""
            />
          </div>
        </div>
      </div>
      <div
        class="sc-fUnMCh lmwSPa"
        title="light-icon"
      >
        <div>
          <div>
            <div
              data-src="theme-sun.svg"
              src=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
